packages/headless-react/src/ssr-commerce/providers.tsx

```diff
@@ -45,30 +50,37 @@ export function buildProviderWithDefinition<
       const {searchActions, controllers} = staticState;
       const hydrateArguments: ControllerPropsMap = {};
 
-      const hydrateControllers: Record<string, object> = {};
-
       for (const [key, controller] of Object.entries(controllers)) {
         const typedController = controller as ControllerWithKind;
 
         switch (typedController._kind) {
           case Kind.Cart:
-            hydrateControllers[key] = {
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any
-              initialState: {items: (controllers as any)[key].state.items},
+            hydrateArguments[key] = {
+              initialState: {
+                items: (controllers as Record<string, Cart>)[key].state.items,
+              },
             };
             break;
 
           case Kind.Context:
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            hydrateControllers[key] = (controllers as any)[key].state;
-
+            hydrateArguments[key] = (controllers as Record<string, Context>)[
```

Comment:
With what I have right now, we will always get a controller with the provided key. No validation is needed. I am looping through every controller, if one of them has a `_kind` that I want, I will get it and use it for my hydrateArguments. Your thrown error will never occur. `controllers[key]` will **always** return something. 

If you are worried with the typing because it is not 100% accurate, sure, I can work on something so the cast looks better but I think that validation has no purpose. It is like adding an `if (false)` statement.

Reply:
Made the cast cleaner with a getController method